### PR TITLE
Remove mocha exit option

### LINKED ISSUE #2673: Investigate what keeps Mocha from exiting
#2672 adds a .mocharc.json file that sets `exit` to `true`. This is because without it, the tests would hang. See https://mochajs.org/#-exit for a more detailed description of the option.

Investigate why the tests are hanging, and remove the option ideally.

### PR DESCRIPTION
**Pull request by TimothyGu**
Fixes #2673.

### CODE REVIEW

**Comment by Sebmaster** [comment]
Shouldn't we close those properly in the after()s instead?

**Comment by TimothyGu** [comment]
That does sound like a better idea perhaps. Let me see if that's doable.

**Comment by domenic** [comment]
I did some local experimenting with a version that actually shuts down the servers instead of using unref, and it's pretty involved. I think we can stick with unref().

**Comment by domenic** [comment]
Hmm, however, waiting for the HTTP servers to shut themselves down causes a 6 second hang at the end, which is not great.

**Comment by domenic** [comment]
OK, I re-did everything. Let's see if it passes on CI...

File: test/web-platform-tests/run-wpts.js [hunk 1]
```diff
@@ -39,13 +39,19 @@ const minimatchers = new Map();
...
 let wptServerURL;
+let serverProcess;
 const runSingleWPT = require("./run-single-wpt.js")(() => wptServerURL);
 before({ timeout: 30 * 1000 }, () => {
-  return startWPTServer({ toUpstream: false }).then(url => {
-    wptServerURL = url;
+  return startWPTServer({ toUpstream: false }).then(({ urls, subprocess }) => {
+    wptServerURL = urls[0];
+    serverProcess = subprocess;
   });
```
Comment (TimothyGu) [inline test/web-platform-tests/run-wpts.js]: Make this an async function?

**Pull request merged**